packages/blade/src/components/InputGroup/_decisions/decisions.md

```diff
@@ -0,0 +1,165 @@
+- [`InputGroup` API](#inputgroup-api)
+- [`InputRow` API](#inputrow-api)
+- [Usage](#usage)
+- [Component Behavior](#component-behavior)
+  - [Validation State Handling](#validation-state-handling)
+  - [Responsive Behavior](#responsive-behavior)
+- [Accessibility](#accessibility)
+
+## Design
+
+[Figma Link](https://www.figma.com/design/jubmQL9Z8V7881ayUD95ps/Blade-DSL?node-id=104468-81766&m=dev) to all variants of the `InputGroup` component.
+
+## Component Hierarchy
+
+The `InputGroup` component is composed of the following components:
+
+```
+InputGroup
+├── InputGroupLabel (internal)
+├── InputRow(s)
+│   └── Children (TextInput, PasswordInput, etc.)
+└── InputGroupHelpText (internal)
+└── InputGroupErrorText (internal)
+```
+
+## `InputGroup` API
+
+The following props are supported by the InputGroup component:
+
+### Prop Table
+
+| Prop            | Type                           | Default  | Description                                                     |
+| --------------- | ------------------------------ | -------- | --------------------------------------------------------------- |
+| label           | string                         | -        | Label for the entire input group.                               |
+| labelPosition   | 'top' \| 'left'                | 'top'    | Position of the label relative to the group.                    |
+| size            | 'medium' \| 'large'            | 'medium' | Controls the size of the input group and its child inputs.      |
+| helpText        | string                         | -        | Help text displayed at the bottom of the group.                 |
+| validationState | 'error' \| 'success' \| 'none' | 'none'   | Controls which validation message is displayed.                 |
+| errorText       | string                         | -        | Error message that appears when validationState is 'error'.     |
+| successText     | string                         | -        | Success message that appears when validationState is 'success'. |
+| isDisabled      | boolean                        | false    | Disables all inputs within the group.                           |
+| children        | React.ReactNode                | -        | Should be InputRow components or other valid inputs.            |
+| testID          | string                         | -        | Test ID for automation.                                         |
+
+## `InputRow` API
+
+### Prop Table
+
+| Prop            | Type            | Default | Description                                                                                                               |
+| --------------- | --------------- | ------- | ------------------------------------------------------------------------------------------------------------------------- |
+| templateColumns | string          | -       | CSS grid template columns value (e.g., "1fr 2fr" or "200px 1fr"). Controls how space is distributed between child inputs. |
```

Comment:
Should we call it `gridTemplateColumns`? makes it consistent with Table component and CSS prop